File: pop_weasel.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jul 28 16:39:40 2023

@author: prsnl
"""
UserInput = input("Please provide an integer: ")

convertNum = int(UserInput)

if converNum > 0:
    

for i in range(1, convertNum+1):
    if i %3 == 0 and i%5 == 0:
        print("BANG")
    elif i%3 == 0:
        print("weasel")
    elif i%5 == 0:
        print("pop")
    else:
        print(i)
        

        

File: pop_weasel.go
package main

import (
	"fmt"
)

var UserInput int

func main() {
	fmt.Println("Please input an integer:")

	fmt.Scanln(&UserInput)
	//fmt.Println("Your number is",UserInput)

	if UserInput < 0 {
		popNeg(UserInput)
	} else if UserInput > 0 {
		popPlus(UserInput)
	} else {
		main()
	}
}

func popNeg(UserInput int) {
	for i := UserInput; i <= 0; i++ {
		if i%5 == 0 && i%3 == 0 {
			fmt.Println("min BANG!")
		} else if i%3 == 0 {
			fmt.Println("min pop")
		} else if i%5 == 0 {
			fmt.Println("min weasel")
		} else {
			fmt.Println(i)
		}
	}
}

func popPlus(UserInput int) {
	for i := 1; i <= UserInput; i++ {
		if i%5 == 0 && i%3 == 0 {
			fmt.Println("BANG!")
		} else if i%3 == 0 {
			fmt.Println("pop")
		} else if i%5 == 0 {
			fmt.Println("weasel")
		} else {
			fmt.Println(i)
		}
	}
}